### PR TITLE
PIC-2325 - BUGFIX Version Mismatch

### PR DESCRIPTION
**Pull request by theDustRoom**
When creating a new report, submitting the Offender Details page caused a version mismatch due to all data being updated for new reports and the session version remained unchanged.

Refactored to only store new data and retain the existing data.

### CODE REVIEW

File: server/controllers/api/apiController.ts [hunk 1]
```diff
@@ -83,11 +83,11 @@ export default class ApiController {
       } else {
         configuredAddress = `${address.buildingName ? convertToTitleCase(address.buildingName) : ''} ${
           address.addressNumber ? address.addressNumber : ''
-        } ${address.streetName ? convertToTitleCase(address.streetName) : ''}
```
Comment (theDustRoom) [inline server/controllers/api/apiController.ts:86]: This tidies up the display of the address

File: server/controllers/shared/sharedController.ts [hunk 2]
```diff
@@ -80,7 +80,7 @@ export default class SharedController {
     if (this.report && this.report.fieldValues && req.session.fieldValues) {
       this.report.fieldValues.forEach(savedValue => {
         const compare = req.session.fieldValues.find(currentValue => currentValue.fieldId === savedValue.fieldId)
-        if (compare && compare.version !== savedValue.version) {
+        if ((compare ? compare.version : 1) !== savedValue.version) {
```
Comment (theDustRoom) [inline server/controllers/shared/sharedController.ts:83]: Make sure we're testing correctly

File: server/services/reportService.ts [hunk 3]
```diff
@@ -54,7 +54,6 @@ export default class ReportService {
       fieldValues.map(async fieldValue => {
         const { reportId, fieldId } = fieldValue
         const foundFieldValue = await getRepository(FieldValue).findOne({ where: { reportId, fieldId } })
-        // @TODO: Enforce field value version - see PIC-1953
```
Comment (theDustRoom) [inline server/services/reportService.ts:57]: Already does elsewhere so removed this TODO

File: server/controllers/api/apiController.ts (hunk 1 above)
Comment (sarahyjja) [reply server/controllers/api/apiController.ts:86]: What's change here out of the linting?

**Review by sarahyjja — APPROVED** [review]

**Pull request merged**